src/AzureIoTHubClient.cpp

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) Arduino. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include <time.h>
+#include <sys/time.h>
+
+#include <AzureIoTHubClient.h>
+
+Client* AzureIoTHubClient::sslClient = NULL;
+
+AzureIoTHubClient::AzureIoTHubClient(Client& sslClient)
+{
+    AzureIoTHubClient::sslClient = &sslClient;
+}
+
+void AzureIoTHubClient::begin()
+{
+#if defined(ARDUINO_ARCH_ESP8266)
+    Serial.setDebugOutput(true);
+#endif
+}
+
+void AzureIoTHubClient::setEpochTime(unsigned long epochTime)
```

Comment:
I was trying to avoid exposing the POSIX time API in the sketch, as this is very un-Arduino like.

Another option is to use a custom [agenttime.h implementation](https://github.com/arduino-libraries/AzureIoT/blob/master/src/sdk/agenttime.h) that is backed  by the ``NTPClient`` library. We can also remove the ```RTCZero``` lib dependency for SAMD. Then the ```AzureIoTHubClient``` constructor or ``begin`` can accept a ```UDP``` instance.

I prefer this approach now, the NTPClient library can auto sync the time at a specified interval. It also make the library much more portable.